platform-modules/content-manager/src/main/java/org/ekstep/content/operation/finalizer/PublishFinalizer.java

```diff
@@ -529,18 +532,36 @@ private void syncNodes(List<Map<String, Object>> children, List<String> unitNode
 				if (CollectionUtils.isNotEmpty(nodeBatch)) {
 					
 					errors = new HashMap<>();
-					Map<String, Object> messages = SyncMessageGenerator.getMessages(nodeBatch, ContentWorkflowPipelineParams.Content.name(), relationMap, errors);
+					List<Map<String, Object>> unitEvents = new ArrayList<>();
+					Map<String, Object> messages = SyncMessageGenerator.getMessages(nodeBatch, ContentWorkflowPipelineParams.Content.name(), relationMap, errors, unitEvents);
 					if (!errors.isEmpty())
 						TelemetryManager.error("Error! while forming ES document data from nodes, below nodes are ignored: " + errors);
 					if(MapUtils.isNotEmpty(messages)) {
-						try {
-                            System.out.println("Number of units to be synced : " + messages.size());
-							ElasticSearchUtil.bulkIndexWithIndexId(ES_INDEX_NAME, DOCUMENT_TYPE, messages);
-                            System.out.println("UnitIds synced : " + messages.keySet());
-						} catch (Exception e) {
-						    e.printStackTrace();
-							TelemetryManager.error("Elastic Search indexing failed: " + e);
-						}					
+						if (IS_ES_SYNC_ENABLED) {
+							try {
+								TelemetryManager.log("Number of units to be synced : " + messages.size());
+								ElasticSearchUtil.bulkIndexWithIndexId(ES_INDEX_NAME, DOCUMENT_TYPE, messages);
+								TelemetryManager.log("UnitIds synced : " + messages.keySet());
+							} catch (Exception e) {
+								e.printStackTrace();
+								TelemetryManager.error("Elastic Search indexing failed: " + e);
+							}
+						}
+					}
+					if(CollectionUtils.isNotEmpty(unitEvents)) {
+						if(StringUtils.isNotBlank(LEARNING_GRAPH_KAFKA_TOPIC)) {
+							unitEvents.stream().forEach(unitEvent -> {
+								try {
+									KafkaClient.send(mapper.writeValueAsString(unitEvent), LEARNING_GRAPH_KAFKA_TOPIC);
+								} catch (Exception e) {
+									e.printStackTrace();
+									TelemetryManager.error("Learning graph topic unit event failed: " + e);
+								}
+							});
```

Comment:
Are we pushing transaction events only here? where is "ov" and "nv" construction here for each property?
Please share one of the transaction events to validate.

Reply:
I shared the transaction event with Amit